From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Wed, 1 Feb 2017 23:49:48 +0100
Subject: [PATCH] {PRINTF} staging/rtl8192e, staging/rtl8192u: fix use of
 sprintf

sprintf() is used in a weird way. This definitely looks like a bug in a
staging driver.
---
 drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c b/drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c
index d6829cf6f7e3..053031bb1425 100644
--- a/drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c
+++ b/drivers/staging/rtl8192u/ieee80211/ieee80211_wx.c
@@ -72,7 +72,9 @@ static inline char *rtl819x_translate_scan(struct ieee80211_device *ieee,
 	iwe.cmd = SIOCGIWNAME;
 	for (i = 0; i < ARRAY_SIZE(ieee80211_modes); i++) {
 		if (network->mode & BIT(i)) {
-			sprintf(pname, ieee80211_modes[i].mode_string, ieee80211_modes[i].mode_size);
+/* BUG?!? */
+			/* sprintf(pname, ieee80211_modes[i].mode_string, ieee80211_modes[i].mode_size); */
+			strcpy(pname, ieee80211_modes[i].mode_string);
 			pname += ieee80211_modes[i].mode_size;
 		}
 	}
-- 
